sdk/highlight-remix/src/client.tsx

```diff
@@ -0,0 +1,30 @@
+import { H } from 'highlight.run'
+import type { HighlightOptions } from 'highlight.run'
+import React, { useEffect } from 'react'
+import { SESSION_STORAGE_KEYS } from '@highlight-run/client/src/utils/sessionStorage/sessionStorageKeys'
+import Cookies from 'js-cookie'
+
+export { H } from 'highlight.run'
+
+interface Props extends HighlightOptions {
+	projectId?: string
+}
+
+export function HighlightInit({ projectId, ...highlightOptions }: Props) {
+	useEffect(() => {
+		if (projectId) {
+			const { sessionSecureID } = H.init(projectId, highlightOptions) || {
+				sessionSecureID: '',
+			}
+
+			if (sessionSecureID) {
+				Cookies.set(
+					SESSION_STORAGE_KEYS.SESSION_SECURE_ID,
+					sessionSecureID,
+				)
+			}
+		}
+	}, []) // eslint-disable-line react-hooks/exhaustive-deps
```

Comment:
should this rerun if `projectID` or `highlightOptions` change?

Reply:
Hmmm. I'd typically not do this, but I'm coming up with some edge cases where our clients may try to mutate these values, in which case we'd need to retry the `H.init`. I'll make the change.